### ISSUE TITLE
missing updated form of filterByType() method in DesignTacoController in chapter 3 in the book

### ISSUE THREAD

**Issue by charlie39** [habuma/spring-in-action-6-samples#11]
After introducing IngredientRepository in chapter 3, the
```
filterByType(List<Ingredient>, Type type)
```
in DesignController.java should be updated to to take an
```Iterable<Ingredient>```
object instead of ```List<Ingredient>```  in given in the sample code. But snippet
for such change is missing from the book.
Should have listing showing the change  right after Listing 3.7

**Comment by AndyLau223** [comment]
Yep, I also found the same issue that's not a big problem. Hence, I change the code as below snippet as work around
  for (final Ingredient.Type type : types) {
            model.addAttribute(type.toString().toLowerCase(),
                    filterByType((**List<Ingredient>)** ingredients,type));
        }

**Comment by charlie39** [comment]
@AndyLau223 there's another issue in page 76 with the ```saveIngredientRefs(long tacoId, List<IngredientRef> ingredientRefs) ```

While calling, ```saveIngredientRefs(tacoId, taco.Inredients()) ; ``` is passed with a ``` List<Ingredient> ``` object while it actually accepts ```List<IngredientRefs>``` object.

**Comment by AndyLau223** [comment]
@charlie39 You need to change the Taco.class entity by making `private List<Ingredient> ingredients = new ArrayList<>();` to ` private List<IngredientRef> ingredients = new ArrayList<>();`

**Comment by KianYang-Lee** [comment]
Leaving below code to stream iterable for anyone that might need it:

```java
    private Iterable<Ingredient> filterByType(Iterable<Ingredient> ingredients, Type type) {
        return StreamSupport.stream(ingredients.spliterator(), false).filter(x -> x.getType().equals(type))
                .collect(Collectors.toList());
```
